hbase_storage/src/main/java/com/splicemachine/si/impl/server/SICompactionStateMutate.java

```diff
@@ -71,11 +72,36 @@ private boolean isSorted(List<Cell> list) {
     }
 
     public void mutate(List<Cell> rawList, List<TxnView> txns, List<Cell> results) throws IOException {
-        assert dataToReturn.isEmpty();
-        assert results.isEmpty();
-        assert maxTombstone == null;
-        assert !LOG.isDebugEnabled() || isSorted(rawList) : "CompactionStateMutate: rawList not sorted";
-        assert rawList.size() == txns.size();
+        final boolean dataToReturnIsEmpty = dataToReturn.isEmpty();
+        final boolean resultsIsEmpty = results.isEmpty();
+        final boolean maxTombstoneIsNull = maxTombstone == null;
+        final boolean rawListAndTxnListSameSize = rawList.size() == txns.size();
+        final boolean debugSortCheck = !LOG.isDebugEnabled() || isSorted(rawList);
+        if (!debugSortCheck) {
+            bypassPurge = true;
+            LOG.warn("Skipping tombstone purge.  rawList is not sorted.");
+        }
+        if (!dataToReturnIsEmpty) {
+            bypassPurge = true;
+            LOG.warn("Skipping tombstone purge.  dataToReturn is not properly initialized.");
+        }
+        if (!resultsIsEmpty) {
+            bypassPurge = true;
+            LOG.warn("Skipping tombstone purge.  results list not properly initialized.");
+        }
```

Comment:
Is this something that happens sometimes or just defensive programming?

Reply:
Yes, this is mainly just defensive programming.  I only ran across 2 of the assertions being hit, but I decided to handle all of them, just in case.  An alternative would be to convert the assert statements into checks which throw IOExceptions.  These would be active on a cluster as well, whether or not assertions were enabled on it.